app/views/properties/show.html.erb

```diff
@@ -115,24 +117,116 @@
+                    <div class="row">
+                        <div class="col-12 mt-2" id="list">
+                            <ul class="list-group list-group-flush">
+                                <li class="list-group-item">Price <span class="float-right">$ <%= @property.price %></span></li>
+                                <li class="list-group-item">Night(s) <span class="float-right" id="totalDays"></span></li>
+                                <li class="list-group-item">Total <span class="float-right" id="totalPrice"></li>
+                            </ul>
+                        </div>
+                        <div class="col-12">
+                            <%= f.submit "Book now", id: "btn-booknow", class: "btn btn-info btn-block mt-4" %>
+                        </div>
+                    </div> <!-- Book Now btn -->
+                </div> <!-- card body -->
+                <% end %>
+            </div> <!-- card -->
         </div> <!-- col 4 -->
-    </div>
+    </div> <!-- row mt-4 -->
 </div> <!-- container -->
+
+<script type="text/javascript">
+    // Datepicker
+    $(function () {
+        $("#start_date").datepicker({
+            minDate: 0,
+            monthNamesShort: ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct",
+                "Nov", "Dec"
+            ],
+            changeMonth: true,
+            changeYear: true,
+            dateFormat: 'mm/dd/yy'
+        });
+        $("#end_date").datepicker({
+            minDate: 0,
+            changeMonth: true,
+            changeYear: true,
+            dateFormat: 'mm/dd/yy'
+        });
+    });
+
+    $(document).ready(() => {
+        $('#list').hide();
+    })
+
+    $('#start_date').change(function () {
+        if (($('#start_date').val() != "") && ($('#end_date').val() != "")) {
+            calculate();
+            $('#btn-booknow').attr('disabled', false);
+            $('#list').show();
+        } else {
+            $('#btn-booknow').attr('disabled', true);
+            $('#list').hide();
+        }
+    });
+
+    $('#end_date').change(function () {
+        if (($('#start_date').val() != "") && ($('#end_date').val() != "")) {
+            calculate();
+            $('#btn-booknow').attr('disabled', false);
+            $('#list').show();
+        } else {
+            $('#btn-booknow').attr('disabled', true);
+            $('#list').hide();
+        }
+    });
+
+    function calculate() {
+        var startDate = new Date($('#start_date').val()),
+            endDate = new Date($('#end_date').val()),
+            daysDiff = calcHolidays(startDate, endDate);
+        if (isNaN(daysDiff)) {
+            $('#totalDays').text("");
+        } else {
+            $('#totalDays').text(daysDiff);
+        }
+
+        var price = parseFloat($('#price').text());
+        var totalPrice = parseFloat(price * daysDiff);
+        $('#totalPrice').text('$ ' + totalPrice + '.0');
+    }
+
+    function calcHolidays(dDate1, dDate2) {
```

Comment:
This can be done simply by subtracting the `start_date` from the `end_date`.
Add in controller when saving total price.

Reply:
This is for the front-end side of the users. This does not actually saved in the database.